### ISSUE TITLE
Merge up to date STMetrics fixes to Mobility test branch.

### ISSUE THREAD

**Issue by BillPascoe** [HughCraig/GHAP#356]
When working in the test environment I notice that some STMetrics fixes that are in production are not in Test. This means test is behind production environment. Update the mobility branch and deploy to test.

**Comment by gdzqzzx** [comment]
That's right. Some fixed issues haven't been done by the time that last stage of mobility development finished. I will merge the latest `dev` later.

**Comment by gdzqzzx** [comment]
In brief, this is caused by conflicts  between the downloaded csv structure and uploaded csv structure.
1. Unaccepted columns like `layer_order` (this also happens in current production on `dataset_order`);
2. Empty values of route related columns from downloaded file. By default, route information would be generated for csv when quantity value is detected, and not all route related columns, for example `route_original_id` and `stop_idx`, are accepted in uploading.

It's unlikely to have a ultimate solution for this bug yet. Highlighting the requirement of accepted mobility dataset and giving clear example for user to replicate would be the best way to ***avoid*** the bug in this stage.
